packages/runtime/src/next/api-handler.ts

```diff
@@ -175,6 +178,30 @@ export function MakeswiftApiHandler(
         return res.json(fonts)
       }
 
+      case 'element-tree': {
+        if (req.query.secret !== apiKey) return res.status(401).json({ message: 'Unauthorized' })
+
+        const elementTree = req.body.elementTree
+        const serializedReplacementContext = req.body.replacementContext
+
+        if (elementTree == null) {
+          return res.status(400).json({ message: 'elementTree must be defined' })
+        }
+
+        if (serializedReplacementContext == null) {
+          return res.status(400).json({ message: 'replacementContext must be defined' })
+        }
+
+        const replacementContext = createReplacementContext(serializedReplacementContext)
```

Comment:
A question I think you might have — "Why don't we need a replacement context endpoint?"

When we were doing templates back in January, we made the choice that we would copy over all **site resources**, and use them to seed the replacement context. So orion can manage that on its own.

In hindsight, I think this was the right call; it's making this work simpler.